index.js

```diff
@@ -29,6 +29,30 @@
   var hasOwnProperty    = Object.prototype.hasOwnProperty;
   var toString          = Object.prototype.toString;
 
+  //  Left :: a -> Either a b
+  var Left = function(x) {
+    return {
+      '@@type': 'sanctuary-def/Either',
+      isLeft: true,
+      isRight: false,
+      chain: function(f) { return this; },
+      map: function(f) { return this; },
+      value: x
+    };
+  };
+
+  //  Right :: b -> Either a b
+  var Right = function(x) {
+    return {
+      '@@type': 'sanctuary-def/Either',
+      isLeft: false,
+      isRight: true,
+      chain: function(f) { return f(this.value); },
+      map: function(f) { return Right(f(this.value)); },
+      value: x
+    };
+  };
```

Comment:
Yep! Simple, principled things make the only reasonable foundation for complex programs. :)